natgeo/modules-analytics
# Feature/syto 150 - Add unit tests

https://jira.nationalgeographic.com/browse/SYTO-150

Good news, everyone!  Unit tests!  :cool: 

```
SUMMARY:
✔ 26 tests completed
-----------------------|-----------|-----------|-----------|-----------|
File                   |   % Stmts |% Branches |   % Funcs |   % Lines |
-----------------------|-----------|-----------|-----------|-----------|
   modules-analytics/  |     89.86 |     72.73 |     82.61 |     91.85 |
      analytics.coffee |     89.86 |     72.73 |     82.61 |     91.85 |
-----------------------|-----------|-----------|-----------|-----------|
All files              |     89.86 |     72.73 |     82.61 |     91.85 |
-----------------------|-----------|-----------|-----------|-----------|

```

[NEW] - Adds Karma running Mocha unit tests with Chai assertions in a PhantomJS environment using Browserify
[UPDATED] - Split grunt tasks into individual files and moved into a separate directory
[UPDATED] - Update module versions in package.json
[UPDATED] - Added to the development section of readme.md to include grunt task descriptions
[RESOLVED] - Fixed coffeelint issues (mainly indention)
[FIXED] - Typo in coffeelint options
[FIXED] - Fixed bug in logic that adds domains to the natGeoDomains list

TO TEST >>>>>
- Check out branch
- `npm install` to update/install dependencies
- `grunt test` to ensure that unit tests run without needing to compile the coffeescript file first.  Tests should pass.
- `grunt dev` to ensure that the dev build process works.  Ensure that analytics.js file remains in the base of the repo.
- `grunt build` to ensure that the production build process works.  Ensure that analytics.js file no longer remains in the base of the repo.

NOTES >>>>>
- I tried not to change much of the main build logic.  We can decide at a later date as a group if we want to use Browserify to compile the files, if we want an unminified version of the module script to stick around (because most modules do), if we want to move the module scripts to dist/src folders, etc.
